completion.go

```diff
@@ -0,0 +1,220 @@
+package cli
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	completionShells = map[string]func(out io.Writer, cmd *cobra.Command) error{
+		"bash": runCompletionBash,
+		"zsh":  runCompletionZsh,
+	}
+	completionShortDescription = "Output shell completion code for the specified shell (bash or zsh)"
+	completionLongDescription  = completionShortDescription + `
+
+Note: this requires the bash-completion framework, which is not installed by default on Mac. This can be installed by using homebrew:
+
+  $ brew install bash-completion
+
+Once installed, bash completion must be evaluated. This can be done by adding the following line to the .bash profile:
+
+  $ source $(brew --prefix)/etc/bash_completion
+
+Note for zsh users: [1] zsh completions are only supported in versions of zsh >= 5.2
+
+Examples:
+  # Load the hcloud completion code for bash into the current shell
+  source <(hcloud completion bash)
+
+  # Load the hcloud completion code for zsh into the current shell
+  source <(hcloud completion zsh)`
+)
+
+func newCompletionCommand(cli *CLI) *cobra.Command {
+	shells := []string{}
+	for s := range completionShells {
+		shells = append(shells, s)
+	}
+
+	cmd := &cobra.Command{
+		Use:       "completion SHELL",
+		Short:     "Output shell completion code for the specified shell (bash or zsh)",
+		Long:      completionLongDescription,
+		RunE:      cli.wrap(runCompletionCommand),
+		ValidArgs: shells,
+	}
+	return cmd
+}
+
+func runCompletionCommand(cli *CLI, cmd *cobra.Command, args []string) error {
+	if len(args) == 0 {
+		return errors.New("shell not specified")
```

Comment:
What about doing validation in a `PreRunE` function like already done in `validateSSHKeyCreate()`?